File: novel-python/app/service/srv_system.py
from app.core import ResponseCode, db, cache, get_global
from app.models import base

@cache.memoize(timeout=3600)
def hot_keywords(num):
    '''
    获取热搜关键词
    '''
    sql = f"SELECT keyword, COUNT(*) cnt FROM sys_keywords GROUP BY keyword ORDER BY COUNT(*) DESC LIMIT {num}"
    results = base.query(sql)
    results = [x['keyword'] for x in results]
    return ResponseCode.Success, results

@cache.memoize(timeout=3600)
def get_categories():
    '''
    获取书籍类目
    '''
    sql = f"SELECT * FROM sys_category WHERE book_cnt > 0"
    results = base.query(sql)
    return ResponseCode.Success, results

File: novel-go/service/server/srv_system.go
package server

import (
	"fmt"
	"novel-go/global"
	"novel-go/model/request"
	"novel-go/model/response"
	"novel-go/model/server"

	"go.uber.org/zap"
)

// 获取热搜关键词
func GetHotKeywords(req *request.Request) (code int, data interface{}) {
	num, _ := req.Data.(map[string]interface{})["num"].(float64)
	if num == 0 {
		num = 30
	}
	keywords := make([]response.Keyword, 0)
	sql := fmt.Sprintf("SELECT keyword, COUNT(*) cnt FROM sys_keywords GROUP BY keyword ORDER BY COUNT(*) DESC LIMIT %d", num)
	err := global.FPG_DB.Raw(sql).Scan(&keywords).Error
	if nil != err {
		global.FPG_LOG.Error("execute sql error", zap.Error(err))
		return response.InternalServerError, nil
	}

	return response.Success, keywords
}

// 获取书籍类目
func GetBookCategories(req *request.Request) (code int, data interface{}) {
	categories := make([]server.ResCategory, 0)
	sql := "SELECT * FROM sys_category WHERE book_cnt > 0"
	err := global.FPG_DB.Raw(sql).Scan(&categories).Error
	if nil != err {
		global.FPG_LOG.Error("execute sql error", zap.Error(err))
		return response.InternalServerError, nil
	}

	return response.Success, categories
}
